torch/csrc/jit/codegen/cuda/scheduler/normalization.cpp

```diff
@@ -344,6 +355,7 @@ std::shared_ptr<ReductionParams> innerPersistentHeuristic(
     // Try to decrease per thread register allocation persistence size on inner
     // reduction
     if (batches_per_block_inner_reduction >= 2 &&
+        batches_per_block_inner_reduction % 2 == 0 &&
```

Comment:
Why is this?

Reply:
we alredy "find bdimx to uniformly distribute workload among batch buffers". If we want to redistribute work load from persistent buffer to bdimx, we must make sure the distribution is uniform. It can avoid cases e.g. changing persistent buffer from 5 to 2.

Reply:
OK, I see. Then why not checking the uniformness directly rather than the divisibility of `batches_per_block_inner_reduction`? It would be more explicit and clear what the heuristics try to balance.